pidtree_bcc/utils.py

```diff
@@ -3,26 +3,29 @@
 import socket
 import struct
 import sys
-from typing import List
+from typing import Generator
 from typing import TextIO
 from typing import Type
 
 import psutil
 
 
-def crawl_process_tree(proc: psutil.Process) -> List[psutil.Process]:
+def crawl_process_tree(pid: int) -> Generator[dict, None, None]:
```

Comment:
Does this mean that the function will be lazily evaluated, and if that's the case will that increase the latency at which the metadata is actually gathered?

Reply:
It won't really make a difference since we are consuming it immediately, but yeah, we'll want a list anyway so might as well just build and return it in the first place, updating.